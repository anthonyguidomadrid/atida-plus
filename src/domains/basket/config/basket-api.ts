export const basketApiConfig = {
  guest: {
    cartUrl: '/guest-carts',
    cartItemsType: 'guest-cart-items',
    multipleCartItemsType: 'multiple-guest-cart-items',
    cartItemsUrl: '/guest-cart-items',
    multipleCartItemsUrl: '/multiple-guest-cart-items'
  },
  authorised: {
    cartUrl: '/carts',
    cartItemsType: 'items',
    multipleCartItemsType: 'multiple-items',
    cartItemsUrl: '/items',
    multipleCartItemsUrl: '/multiple-items'
  }
} as const
