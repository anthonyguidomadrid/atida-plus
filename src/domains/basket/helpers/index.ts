export * from './basket-api'
export * from './get-price-channel'
