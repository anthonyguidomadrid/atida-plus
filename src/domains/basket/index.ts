export * from './normalizers'
export * from './sagas'
export * from './selectors'
export * from './slices'
