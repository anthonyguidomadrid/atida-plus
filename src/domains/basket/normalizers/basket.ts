import { removeUndefinedPropertiesFromObject } from '~helpers'
import { getPriceChannel } from '../helpers'
import {
  SprykerBasket,
  Basket,
  SprykerBasketItem,
  BasketItem,
  BasketExpense,
  PromotionalItem,
  SessionChannelType
} from '../types'

export const normalizeBasket = (
  basket?: Partial<SprykerBasket>,
  useMockedData?: SessionChannelType
): Basket =>
  removeUndefinedPropertiesFromObject({
    id: basket?.data?.id,
    priceChannel: getPriceChannel(basket?.included, useMockedData),
    currency: basket?.data?.attributes?.currency,
    items: basket?.included
      ?.filter(({ type }) => type === 'guest-cart-items' || type === 'items')
      .map(normalizeBasketItem),
    subTotal: basket?.data?.attributes?.totals?.subtotal,
    itemTotal: basket?.data?.attributes?.totals?.itemTotal,
    shippingTotal: normalizeBasketExpenses(
      basket?.data?.attributes?.expenses,
      basket?.data?.attributes?.totals.expenseTotal
    ), // TODO: might need to be more specific like Pure
    grandTotal: basket?.data?.attributes?.totals?.priceToPay,
    taxTotal: basket?.data?.attributes?.totals?.taxTotal,
    surchargeTotal: basket?.data?.attributes?.totals?.surchargeTotal,
    rrpTotal: basket?.data?.attributes?.totals?.subtotal, // TODO: this is probably wrong
    discountTotal: basket?.data?.attributes?.totals?.discountTotal,
    rrpDiscountTotal: basket?.data?.attributes?.totals?.rrpDiscountTotal,
    totalSaving: basket?.data?.attributes?.totals?.totalSaving,
    rewardTotal: basket?.data?.attributes?.totals?.rewardTotal,
    freeShippingThreshold:
      basket?.data?.attributes?.totals?.freeShippingThreshold,
    coupons: basket?.data?.attributes?.productDiscounts?.filter(
      item => item.code !== null
    ),
    discounts: basket?.data?.attributes?.productDiscounts?.filter(
      item => item.code === null
    ),
    cartCoupons: basket?.data?.attributes?.discounts?.filter(
      item => item.code !== null
    ),
    expenses: basket?.data?.attributes?.expenses,
    promotionalItems: basket?.included
      ?.filter(
        ({ type, attributes }) =>
          type === 'promotional-items' &&
          attributes.discountPercentage === 10000
      )
      .map(normalizeBasketPromotionalItem),
    discountedItems: basket?.included
      ?.filter(
        ({ type, attributes }) =>
          type === 'promotional-items' &&
          attributes?.discountPercentage &&
          attributes.discountPercentage < 10000
      )
      .map(normalizeBasketPromotionalItem),
    deliveryDays: basket?.data?.attributes?.deliveryDays
  })

const normalizeBasketItem = (item?: SprykerBasketItem): BasketItem =>
  removeUndefinedPropertiesFromObject({
    id: item?.id,
    sku: item?.attributes?.sku,
    quantity: item?.attributes?.quantity,
    unitPrice: item?.attributes?.calculations?.unitPrice,
    subtotal: item?.attributes?.calculations?.sumPrice,
    tax: item?.attributes?.calculations?.unitTaxAmountFullAggregation,
    discount: item?.attributes?.calculations?.unitDiscountAmountAggregation,
    unitPriceToPayAggregation:
      item?.attributes?.calculations?.unitPriceToPayAggregation,
    isPromo: item?.attributes?.isPromo,
    isFullyDiscounted: item?.attributes?.isFullyDiscounted,
    isPrescription: item?.attributes?.isPrescription,
    tokenCount: item?.attributes?.tokenCount,
    priceChannel: item?.attributes.price_channel
  })

const normalizeBasketExpenses = (
  expenses?: BasketExpense[],
  shippingTotal?: number
): number | undefined => {
  if (!Array.isArray(expenses)) return shippingTotal

  return expenses.reduce(
    (memo, expense) =>
      memo + expense.sumPriceToPayAggregation * expense.quantity,
    0
  )
}

const normalizeBasketPromotionalItem = (
  item?: SprykerBasketItem
): PromotionalItem => {
  return removeUndefinedPropertiesFromObject({
    id_promotional_item: item?.id,
    sku: item?.attributes?.sku.replace(/\D/g, ''),
    quantity: item?.attributes?.quantity ?? 1,
    is_promo: true,
    userDeclined: item?.attributes?.userDeclined,
    isNew: item?.attributes?.isNew,
    discountPercentage: item?.attributes?.discountPercentage
  })
}
