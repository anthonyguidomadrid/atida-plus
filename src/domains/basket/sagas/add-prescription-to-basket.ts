import type { PayloadAction } from '@reduxjs/toolkit'
import { AxiosResponse } from 'axios'
import type { SagaIterator } from 'redux-saga'
import {
  all,
  call,
  getContext,
  put,
  select,
  takeEvery
} from 'redux-saga/effects'
import { createClient, logger } from '~helpers'
import {
  addPrescriptionToBasketTrigger,
  addPrescriptionToBasketRequest,
  addPrescriptionToBasketSuccess,
  addPrescriptionToBasketFailure,
  addPrescriptionToBasketFulfill
} from '../slices'
import { selectId } from '~domains/basket'
import { ApiError, getErrorMessage } from '~helpers/error'
import { BasketWithProducts } from '../types'
import { isJson } from '~helpers/isJson'

function* doAddPrescriptionToBasket({
  payload
}: PayloadAction<string | string[]>) {
  try {
    const locale: string = yield getContext('locale')
    const client = createClient({ locale, addAnonymousCustomerUniqueId: true })
    const cartId: string = yield select(selectId)

    yield put(addPrescriptionToBasketRequest())
    const response: AxiosResponse<BasketWithProducts> = yield call(
      client.post,
      '/api/basket/add-prescription',
      { cartId, prescriptionToken: payload }
    )

    yield put(addPrescriptionToBasketSuccess(response.data))
  } catch (error) {
    const errorMessages = getErrorMessage(error)
    let errorDetail = (error as ApiError)?.response?.data?.details?.detail
    errorDetail = isJson(errorDetail) ? JSON.parse(errorDetail) : errorDetail
    logger.error(
      {
        message: getErrorMessage(error),
        name: getErrorMessage(error)
      },
      JSON.stringify({
        payload,
        errorDetail
      })
    )
    if (Array.isArray(errorMessages)) {
      yield put(
        addPrescriptionToBasketFailure({
          message: errorMessages[0]
        })
      )
    } else {
      yield put(
        addPrescriptionToBasketFailure({
          message: errorMessages
        })
      )
    }
  } finally {
    yield put(addPrescriptionToBasketFulfill())
  }
}
export function* addPrescriptionToBasketSaga(): SagaIterator {
  yield all([
    takeEvery(addPrescriptionToBasketTrigger, doAddPrescriptionToBasket)
  ])
}
