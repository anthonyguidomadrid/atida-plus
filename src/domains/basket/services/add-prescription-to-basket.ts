import getConfig from 'next/config'

import { createClient } from '~helpers'

import type { Basket, SprykerBasket } from '../types'
import { normalizeBasket } from '../normalizers'
import { getBasketApiConfig } from '../helpers'
import { fetchProductsForBasket } from './fetch-products-for-basket'

if (typeof window !== 'undefined')
  throw new Error('This file should not be called client-side')

export const addPrescriptionToBasket = async (
  locale: string,
  user: {
    anonymousCustomerUniqueId?: string
    token?: string
  },
  prescriptionToken?: string | string[],
  cartId?: string
): Promise<Basket> => {
  const { serverRuntimeConfig } = getConfig()
  const client = createClient({
    locale,
    options: {
      baseURL: serverRuntimeConfig.ecommerceHost[locale],
      headers: user.token
        ? {
            Authorization: `Bearer ${user.token}`
          }
        : undefined
    },
    addAnonymousCustomerUniqueId: user.token
      ? undefined
      : user.anonymousCustomerUniqueId
  })
  const apiConfig = getBasketApiConfig(user.token)

  const requestUrl = user.token
    ? `${apiConfig.cartUrl}/${cartId}/prescription-tokens?include=items,promotional-items,abstract-product,cart-rules,vouchers`
    : apiConfig.cartUrl
  const response = await client.post<SprykerBasket>(requestUrl, {
    data: {
      type: 'prescription-tokens',
      attributes: {
        tokens: Array.isArray(prescriptionToken)
          ? prescriptionToken
          : [prescriptionToken]
      }
    }
  })

  const basket = normalizeBasket(response.data)

  const productsInBasket = await fetchProductsForBasket(locale, basket)

  return productsInBasket
}
