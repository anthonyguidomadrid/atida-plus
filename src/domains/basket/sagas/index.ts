import type { SagaIterator } from 'redux-saga'
import { all, fork } from 'redux-saga/effects'
import { addToBasketSaga } from './add-to-basket'
import { addSeveralToBasketSaga } from './add-several-to-basket'
import { getBasketSaga } from './get-basket'
import { removeFromBasketSaga } from './remove-from-basket'
import { changeItemQuantitySaga } from './change-item-quantity'
import { addCouponSaga } from './add-coupon'
import { removeCouponSaga } from './remove-coupon'
import { addPrescriptionToBasketSaga } from './add-prescription-to-basket'

export function* basketSaga(): SagaIterator {
  yield all(
    [
      addToBasketSaga,
      addSeveralToBasketSaga,
      removeFromBasketSaga,
      changeItemQuantitySaga,
      getBasketSaga,
      addCouponSaga,
      addPrescriptionToBasketSaga,
      removeCouponSaga
    ].map(fork)
  )
}
