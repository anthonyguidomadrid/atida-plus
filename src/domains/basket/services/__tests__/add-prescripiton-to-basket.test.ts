/**
 * @jest-environment node
 */
import axios from 'axios'
// @ts-ignore
import { mget } from '@elastic/elasticsearch'
import { addPrescriptionToBasket } from '../add-prescription-to-basket'
import { basketApiConfig } from '../../config'
import { sprykerBasket, elasticsearchProducts } from '../../__mocks__/basket'
import { createClient } from '~helpers'

describe(addPrescriptionToBasket, () => {
  beforeEach(() => {
    ;(axios.post as jest.Mock).mockResolvedValue({ data: sprykerBasket })
  })

  it('creates the client & passes the locale and token to spryker request', async () => {
    await addPrescriptionToBasket(
      'en-gb',
      {
        anonymousCustomerUniqueId: 'some-unique-id',
        token: 'some-user-token'
      },
      'some token',
      'cart-id'
    )
    expect(createClient).toHaveBeenCalledWith({
      addAnonymousCustomerUniqueId: undefined,
      locale: 'en-gb',
      options: {
        baseURL: 'spryker.co.uk',
        headers: {
          Authorization: `Bearer some-user-token`
        }
      }
    })
    expect(axios.post).toHaveBeenCalledWith(
      `${basketApiConfig.authorised.cartUrl}/cart-id/prescription-tokens?include=items,promotional-items,abstract-product,cart-rules,vouchers`,
      {
        data: {
          type: 'prescription-tokens',
          attributes: {
            tokens: ['some token']
          }
        }
      }
    )
  })

  it('returns the normalized basket response', async () => {
    mget.mockResolvedValue(elasticsearchProducts)
    const response = await addPrescriptionToBasket(
      'en-gb',
      {
        anonymousCustomerUniqueId: 'some-unique-id',
        token: 'some-user-token'
      },
      'some token',
      'cart-id'
    )
    expect(response).toMatchSnapshot()
  })
})
