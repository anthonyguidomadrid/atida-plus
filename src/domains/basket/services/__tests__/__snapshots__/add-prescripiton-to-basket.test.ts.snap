// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`addPrescriptionToBasket returns the normalized basket response 1`] = `
Object {
  "cartCoupons": Array [],
  "coupons": Array [],
  "currency": "EUR",
  "discountTotal": 0,
  "discountedItems": Array [],
  "discounts": Array [],
  "expenses": Array [
    Object {
      "quantity": 1,
      "sumPriceToPayAggregation": 399,
      "type": "SHIPMENT_EXPENSE",
      "unitGrossPrice": 0,
      "unitNetPrice": 0,
    },
  ],
  "freeShippingThreshold": 5900,
  "grandTotal": 9999,
  "id": "f0849e9c-6712-54b9-8db5-440438d03387",
  "itemTotal": 648,
  "items": Array [
    Object {
      "discount": 999,
      "id": "100000001",
      "isPromo": false,
      "product": Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
        },
        "availability": "AVAILABLE",
        "brand": Object {
          "code": "atida",
          "label": "Atida",
        },
        "categories": Object {
          "lvl0": Array [
            "main",
          ],
        },
        "contentSizeFactor": 1,
        "description": "<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>
</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!
</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.
</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.
</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.
</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...
</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>
</p>",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "gtin": "8428148462962",
        "id": "100000001",
        "images": Array [
          Object {
            "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
            "productHigh": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
            "productTileThumbnail": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
          },
        ],
        "labels": Array [
          Object {
            "label": "Free sponge",
            "type": "promotion",
          },
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "manufacturerName": "LOreal Deutschland",
        "maxQuantity": 50,
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "metaDescription": "Meta description No contiene gluten. Apto para personas celíacas Meta description",
        "metaKeywords": "Meta keywords NaturGreen Algarroba Meta keywords",
        "metaTitle": "Meta title NaturGreen Algarroba a la Taza BIO 225gr Meta title",
        "name": "Test Product 1",
        "netWeight": Object {
          "amount": 100,
          "unit": "Grams",
          "unitLabel": "g",
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "pzn": "1234567",
        "rating": Object {
          "averageRating": 3.5,
          "numberOfReviews": 100,
        },
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "shortDescription": "Preparado vegetal con harina de algarroba ecológica, ideal para desayunos y meriendas",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
      },
      "quantity": 1,
      "sku": "100000001",
      "subtotal": 9999,
      "tax": 150,
      "unitPrice": 9999,
      "unitPriceToPayAggregation": 9849,
    },
    Object {
      "discount": 999,
      "id": "100000002",
      "isPromo": false,
      "product": Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
        },
        "availability": "AVAILABLE",
        "brand": Object {
          "code": "atida",
          "label": "Atida",
        },
        "categories": Object {
          "lvl0": Array [
            "main",
          ],
        },
        "contentSizeFactor": 1,
        "description": "<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>
</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!
</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.
</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.
</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.
</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...
</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>
</p>",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "gtin": "8428148462962",
        "id": "100000002",
        "images": Array [
          Object {
            "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
            "productHigh": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
            "productTileThumbnail": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
          },
        ],
        "labels": Array [
          Object {
            "label": "Free sponge",
            "type": "promotion",
          },
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "manufacturerName": "LOreal Deutschland",
        "maxQuantity": 50,
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "metaDescription": "Meta description No contiene gluten. Apto para personas celíacas Meta description",
        "metaKeywords": "Meta keywords NaturGreen Algarroba Meta keywords",
        "metaTitle": "Meta title NaturGreen Algarroba a la Taza BIO 225gr Meta title",
        "name": "Test Product 2",
        "netWeight": Object {
          "amount": 100,
          "unit": "Grams",
          "unitLabel": "g",
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "pzn": "1234567",
        "rating": Object {
          "averageRating": 3.5,
          "numberOfReviews": 100,
        },
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "shortDescription": "Preparado vegetal con harina de algarroba ecológica, ideal para desayunos y meriendas",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
      },
      "quantity": 7,
      "sku": "100000002",
      "subtotal": 9999,
      "tax": 150,
      "unitPrice": 9999,
      "unitPriceToPayAggregation": 9849,
    },
    Object {
      "discount": 999,
      "id": "100000003",
      "isPromo": false,
      "product": Object {
        "alternativeUrls": Object {
          "es-es": "",
          "pt-pt": "forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
        },
        "availability": "AVAILABLE",
        "brand": Object {
          "code": "atida",
          "label": "Atida",
        },
        "categories": Object {
          "lvl0": Array [
            "main",
          ],
        },
        "contentSizeFactor": 1,
        "description": "<p><strong>Forcapil Anti-Caída del Cabello Arkopharma 30 Cápsulas 2+1</strong>
</p><p>¡Recupera la fuerza y vitalidad de tu cabello con Forcapil!
</p><p>Indicado para hombres y mujeres, este nutricosmético previene y combate la caída del pelo.
</p><p>Fórmula exclusivamente con ingredientes 100% de origen natural.
</p><p>Tiene una alta concentración en biotina, y su composición con aminoácidos, vitaminas, minerales y plantas favorece el crecimiento de la melena.
</p><p>Se recomienda tomar en momentos en los que sufrimos una mayor pérdida de cabello, causada por estrés, menopausia, tras el parto, cambios de estación...
</p><p><strong>Tratamiento para 3 meses: 2 + 1 de REGALO.</strong>
</p>",
        "enabled": true,
        "format": Object {
          "code": "shampooCode",
          "label": "Shampoo",
        },
        "gtin": "8428148462962",
        "id": "100000003",
        "images": Array [
          Object {
            "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
            "productHigh": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
            "productTileThumbnail": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
          },
        ],
        "labels": Array [
          Object {
            "label": "Free sponge",
            "type": "promotion",
          },
          Object {
            "label": "Outlet",
            "type": "pricing-tag-outlet",
          },
          Object {
            "label": "-11%",
            "type": "discount",
          },
        ],
        "largeImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20High-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "manufacturerName": "LOreal Deutschland",
        "maxQuantity": 50,
        "mediumImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Medium-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "metaDescription": "Meta description No contiene gluten. Apto para personas celíacas Meta description",
        "metaKeywords": "Meta keywords NaturGreen Algarroba Meta keywords",
        "metaTitle": "Meta title NaturGreen Algarroba a la Taza BIO 225gr Meta title",
        "name": "Test Product 3",
        "netWeight": Object {
          "amount": 100,
          "unit": "Grams",
          "unitLabel": "g",
        },
        "pricePerUnit": Object {
          "currency": "EUR",
          "unit": "100 ml",
          "value": 8888,
        },
        "productDatImage": "https://sandbox-atida.bynder.com/transform/db2e62c0-2c32-4e50-bee7-1a7a421323f1/ean_img_8833772266123_0",
        "productTileImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "productTileRetinaImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Retina-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "pzn": "1234567",
        "rating": Object {
          "averageRating": 3.5,
          "numberOfReviews": 100,
        },
        "rrp": Object {
          "currency": "EUR",
          "value": 9999,
        },
        "shortDescription": "Preparado vegetal con harina de algarroba ecológica, ideal para desayunos y meriendas",
        "thumbnailImage": "https://d2csxpduxe849s.cloudfront.net/media/07EADC8C-9435-4B3D-87CF5EDA882A36F2/F1974216-C01C-415C-B00F4AEBC6928147/Product%20Tile%20Thumbnail-51236DF8-2D7D-4757-AF2614FABA305C0A.png",
        "unitVolume": Object {
          "amount": 100,
          "unit": "MILLILITER",
          "unitLabel": "ml",
        },
        "url": "/forcapil-anti-caida-del-cabello-arkopharma-30-capsulas-2+1",
      },
      "quantity": 3,
      "sku": "100000003",
      "subtotal": 9999,
      "tax": 150,
      "unitPrice": 9999,
      "unitPriceToPayAggregation": 9849,
    },
  ],
  "promotionalItems": Array [],
  "rrpTotal": 9999,
  "shippingTotal": 399,
  "subTotal": 9999,
  "surchargeTotal": 123,
  "taxTotal": 1596,
}
`;
